diff --git a/CMakeLists.txt b/CMakeLists.txt
index de4e1509..6f5f2ad3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,6 @@
 cmake_minimum_required(VERSION 2.8.8)
 project(czmq)
 enable_language(C)
-enable_testing()
 
 set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
@@ -113,6 +112,10 @@ if (CYGWIN)
     set(MORE_LIBRARIES)
 endif()
 
+if (UNIX AND NOT APPLE)
+    set(MORE_LIBRARIES m)
+endif()
+
 list(APPEND CMAKE_MODULE_PATH "${SOURCE_DIR}")
 set(OPTIONAL_LIBRARIES)
 set(OPTIONAL_LIBRARIES_STATIC)
@@ -546,27 +549,6 @@ target_link_libraries(
     ${OPTIONAL_LIBRARIES_STATIC}
 )
 endif()
-add_executable(
-    test_randof
-    "${SOURCE_DIR}/src/test_randof.c"
-)
-if (TARGET czmq)
-target_link_libraries(
-    test_randof
-    czmq
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-)
-endif()
-if (NOT TARGET czmq AND TARGET czmq-static)
-target_link_libraries(
-    test_randof
-    czmq-static
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-    ${OPTIONAL_LIBRARIES_STATIC}
-)
-endif()
 add_executable(
     czmq_selftest
     "${SOURCE_DIR}/src/czmq_selftest.c"
@@ -656,21 +638,6 @@ IF (ENABLE_DRAFTS)
 ENDIF (ENABLE_DRAFTS)
 
 
-add_custom_target(
-    copy-selftest-ro ALL
-    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_SOURCE_DIR}/src/selftest-ro ${CMAKE_BINARY_DIR}/src/selftest-ro
-)
-
-add_custom_target(
-    make-selftest-rw ALL
-    COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/src/selftest-rw
-)
-
-set_directory_properties(
-    PROPERTIES
-    ADDITIONAL_MAKE_CLEAN_FILES "${CMAKE_BINARY_DIR}/src/selftest-ro;${CMAKE_BINARY_DIR}/src/selftest-rw"
-)
-
 foreach(TEST_CLASS ${TEST_CLASSES})
     add_test(
         NAME ${TEST_CLASS}
@@ -707,7 +674,6 @@ set(cmake_generated ${CMAKE_BINARY_DIR}/CMakeCache.txt
                     ${CMAKE_BINARY_DIR}/src/czmq_selftest
                     ${CMAKE_BINARY_DIR}/src/zmakecert
                     ${CMAKE_BINARY_DIR}/src/zsp
-                    ${CMAKE_BINARY_DIR}/src/test_randof
                     ${CMAKE_BINARY_DIR}/src/czmq_selftest
 )
 